src/NLog/GlobalDiagnosticsContext.cs

```diff
@@ -57,22 +57,42 @@ public static void Set(string item, string value)
         }
 
         /// <summary>
+        /// Sets the Global Diagnostics Context item to the specified value.
+        /// </summary>
+        /// <param name="item">Item name.</param>
+        /// <param name="value">Item value.</param>
+        public static void Set(string item, object value)
+        {
+            lock (dict)
+            {
+                dict[item] = value;
+            }
+        }
+
+        /// <summary>
         /// Gets the Global Diagnostics Context named item.
         /// </summary>
         /// <param name="item">Item name.</param>
         /// <returns>The item value of string.Empty if the value is not present.</returns>
         public static string Get(string item)
         {
+            return RenderContextObject(GetObject(item));
```

Comment:
I think this is not fully backwards compatible?

a item with the value `null` now returns `null` and previously `string.Empty`?